### PR TITLE
Limit array size to 2^32 

### PR DESCRIPTION
**Pull request by cpq**
Hi @alashkin, please make a review! Since @mmikulicic is off today, let this PR take over PR #336

### CODE REVIEW

File: src/vm.c [hunk 1]
```diff
@@ -732,12 +732,11 @@ V7_PRIVATE val_t v7_property_value(struct v7 *v7, val_t obj,
...
+V7_PRIVATE unsigned long v7_array_length(struct v7 *v7, val_t v) {
   struct v7_property *p;
-  long max = -1, k;
+  unsigned long max = 0, k;
   char *end;
 
-  (void) v7;
   if (!v7_is_object(v)) {
     return -1;
   }
```
Comment (alashkin) [inline src/vm.c]: How `return -1` will work with unsigned return type?

Comment (alashkin) [inline src/vm.c]: I mean, it gives 0xFFFFFFF...... and will be lenght number on 32b.

**Comment by cpq** [comment]
PTAL

File: src/vm.c (hunk 1 above)
Comment (cpq) [inline src/vm.c]: Good catch. Fixed (returning 0).

**Comment by alashkin** [comment]
LGTM

**Pull request merged**